### ISSUE TITLE
Issue with word_delimiter and minimum_should_match

### ISSUE THREAD

**Issue by romainruaud** [Smile-SA/elasticsuite#3173]
### Steps to reproduce

1. Create a product named PRODUCT
2. Ensure minimum_should_match is set to 100% (default)
3. Search for ProDuct

### Expected result

1. The product named PRODUCT shall appear

### Actual result

1. No results

The product is excluded by this part of the exact query : 

```
          "filter": {
            "multi_match": {
              "query": "ProDuct",
              "fields": [
                "search^1",
                "sku^1"
              ],
              "minimum_should_match": "100%",
              "tie_breaker": 1,
              "boost": 1,
              "type": "best_fields",
              "cutoff_frequency": 0.15
            }
          },
```

**Comment by rbayet** [comment]
TL;DR version: the multi_match does not handle correctly a minimum_should_match at 100% with queries that, once analyzed, might produce several tokens at the same place.
Which will be the case for "ProDuct" which will generate "product" (pos=0), "pro" (pos=0) and "duct" (pos=1).

Solution: Switching "split_on_case_change" to "false" in the definition of the "world_delimiter" filter in etc/elasticsuite_analysis.xml (see below) should do the trick, while losing the (dubious) benefit of being able to handle the "StarWars/starwars/star-wars" case.
```
        <filter name="word_delimiter" type="word_delimiter" language="default">
            <generate_word_parts>true</generate_word_parts>
            <catenate_words>true</catenate_words>
            <catenate_numbers>true</catenate_numbers>
            <catenate_all>true</catenate_all>
            <split_on_case_change>true</split_on_case_change> <== change this to false, clear caches and reindex
            <split_on_numerics>true</split_on_numerics>
            <preserve_original>true</preserve_original>
        </filter>
```

Enabling the experimental Search Relevance settings "Spellchecking Configuration > Term Vectors Configuration
> [Experimental] Use all tokens from term vectors" is not guaranteed to provide any help, on the contrary in such case.
